pkg/scheduler/api/pod_info.go

```diff
@@ -61,15 +61,36 @@ import (
 func GetPodResourceRequest(pod *v1.Pod) *Resource {
 	result := GetPodResourceWithoutInitContainers(pod)
 
-	// take max_resource(sum_pod, any_init_container)
+	// Calculate the sum of CPU and Memory resources used by containers
+	var cpuSum, memorySum float64
+	for _, container := range pod.Spec.Containers {
+		cpuSum += float64(container.Resources.Requests.Cpu().MilliValue())
+		memorySum += float64(container.Resources.Requests.Memory().Value())
+	}
+
+	// Calculate max utilization from init containers
+	var initContainerMax float64
 	for _, container := range pod.Spec.InitContainers {
-		result.SetMaxResource(NewResource(container.Resources.Requests))
+		containerResource := NewResource(container.Resources.Requests)
+		initContainerMax = math.Max(initContainerMax, containerResource.MilliCPU)
+		initContainerMax = math.Max(initContainerMax, float64(containerResource.Memory))
+	}
+
+	// Adjust the share based on CPU, Memory, and maximum init container utilization
+	maxUtilization := math.Max(cpuSum, memorySum)
+	maxUtilization = math.Max(maxUtilization, initContainerMax)
+	if maxUtilization > 0 {
+		podShare := maxUtilization / 1000 // Convert to fractional value
+		if podShare > 1 {
+			podShare = 1
+		}
+		result.AddScalar(v1.ResourcePods, podShare)
```

Comment:
Aha, seems it's misunderstanding, we assume the value 1 as scaler resource, which represents pod itself occupied a pod resource and this is used in preemption scenario, here the value should always be 1, and we should fix it in proportion plugin instead: )

Reply:
Thank you much for your review!!!! I am working on it. I thought we need to adjust the resources accordingly. Can you provide me some more hints on the issue.